source/Crew.cpp

```diff
@@ -0,0 +1,242 @@
+			else if(child.Token(0) == "minimum per ship")
+				minimumPerShip = max((int)child.Value(1), 0);
+			// The number of credits paid daily while parked (minimum 0)
+			else if(child.Token(0) == "parked salary")
+				parkedSalary = max((int)child.Value(1), 0);
+			// Every nth crew member on the ship will be this crew member
+			else if(child.Token(0) == "population per member")
+				populationPerMember = max((int)child.Value(1), 0);
+			// The number of credits paid daily (minimum 0)
+			else if(child.Token(0) == "salary")
+				salary = max((int)child.Value(1), 0);
+			else
+				child.PrintTrace("Skipping unrecognized attribute:");
+		}
+		// If true, the crew member will not appear on escorts
+		else if(child.Token(0) == "avoids escorts")
+			avoidsEscorts = true;
+		// If true, the crew member will not appear on the flagship
+		else if(child.Token(0) == "avoids flagship")
+			avoidsFlagship = true;
+		else
+			child.PrintTrace("Skipping incomplete attribute:");
+	}
+}
+
+
+
+int64_t Crew::CalculateSalaries(
+	const vector<shared_ptr<Ship>> &ships,
+	const Ship * &flagship,
+	const bool includeExtras
+)
+{
+	int64_t totalSalaries = 0;
+	
+	for(const shared_ptr<Ship> &ship : ships)
+	{
+		totalSalaries += Crew::SalariesForShip(
+			ship,
+			ship.get() == flagship,
+			includeExtras
+		);
+	}
+	
+	return totalSalaries;
+}
+
+
+
+int64_t Crew::CostOfExtraCrew(
+	const vector<shared_ptr<Ship>> &ships,
+	const Ship * &flagship
+)
+{
+	// Calculate with and without extras and return the difference.
+	return Crew::CalculateSalaries(ships, flagship, true)
+		- Crew::CalculateSalaries(ships, flagship, false);
+}
+
+
+
+int64_t Crew::NumberOnShip(
+	const Crew &crew,
+	const shared_ptr<Ship> &ship,
+	const bool isFlagship,
+	const bool includeExtras
+)
+{
+	int64_t count = 0;
+	
+	// If this is the flagship, check if this crew avoids the flagship.
+	if(isFlagship && crew.AvoidsFlagship())
+		return count;
+	// If this is an escort, check if this crew avoids escorts.
+	if(!isFlagship && crew.AvoidsEscorts())
+		return count;
+	
+	const int64_t countableCrewMembers = includeExtras
+		? ship->Crew()
+		: ship->RequiredCrew();
+	
+	// Apply the per-ship minimum.
+	 count = min(crew.MinimumPerShip(), countableCrewMembers);
+	
+	// Prevent division by zero so that the universe doesn't implode.
+	if(crew.PopulationPerMember())
+	{
+		// Figure out how many of this kind of crew we have, by population.
+		count = max(
+			count,
+			countableCrewMembers / crew.PopulationPerMember()
+		);
+	}
+	
+	return count;
+}
+
+
+
+int64_t Crew::SalariesForShip(
+	const shared_ptr<Ship> &ship,
+	const bool isFlagship,
+	const bool includeExtras
+)
+{
+	// We don't need to pay dead people.
+	if(ship->IsDestroyed())
+		return 0;
+	
+	int64_t salariesForShip = 0;
+	int64_t specialCrewMembers = 0;
+	
+	// Add up the salaries for all of the special crew members
+	for(const pair<const string, Crew>& crewPair : GameData::Crews())
```

Comment:
It is this repo's style to attach ointer or reference signifier to the variable and not the type, e.g. `int *i` and not `int* i`